samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/evaluation/HistogramListener.java

```diff
@@ -2448,14 +2448,16 @@ private Map getAssessmentStatisticsMap(ArrayList scoreList)
 
     if (doubles.isEmpty())
       doubles.add(new Double(0.0));
-    Object[] array = doubles.toArray();
-    Arrays.sort(array);
+    Object[] scoresSorted = doubles.toArray();
+    Arrays.sort(scoresSorted);
 
-    double[] scores = new double[array.length];
-    for (int i=0; i<array.length; i++)
-{
-      scores[i] = Double.valueOf(castingNum((Double)array[i],2)).doubleValue();
-}
+    double[] scores = new double[scoresSorted.length];
+
+    for (int i=0; i<scoresSorted.length; i++){
```

Comment:
Can you do simpler format instead?

for (Double d : scoresSorted) {

Can you add generics to avoid the casts?